### ISSUE TITLE
Faceted tag search for plugins and tools

### ISSUE THREAD

**Issue by simonw** [simonw/datasette.io#91]
I want to be able to see all of the geospatial plugins, or all of the plugins that use a specific hook.

**Comment by simonw** [comment]
For plugin hooks: I could try installing all of the plugins in a virtual environment and then run `datasette --get /-/plugins.json` to see what hooks they use.

**Comment by simonw** [comment]
I tried that trick and it failed, because two of the current batch of plugins cause Datasette to crash on startup if they aren't configured properly:

- `datasette-auth-existing-cookies`
- `datasette-redirect-to-https`

After uninstalling those to (and installing everything in `plugin_repos.yaml`) I got the following output: https://gist.github.com/simonw/f95207c1483fe74990af5a92ef447315

**Comment by simonw** [comment]
Here's what I did:

```
python -m venv plugins-venv
# Convert plugin_repos.yml to all-plugins.txt by hand
plugins-venv/bin/pip install -r all-plugins.txt
plugins-venv/bin/datasette --get '/-/plugins.json'
# This broke
plugins-venv/bin/pip uninstall datasette-auth-existing-cookies
plugins-venv/bin/pip uninstall datasette-redirect-to-https
# Now this worked:
plugins-venv/bin/datasette --get '/-/plugins.json' | jq
```

**Comment by simonw** [comment]
I'm tempted to run a version of the above in a separate repo as a git scraper, which then commits that plugins.json file so that the `datasette.io` build script can load it and use it to populate a table.

**Comment by simonw** [comment]
This came up again on Discord: https://discord.com/channels/823971286308356157/823971286941302908/1004879739862909058

@davidbgk made a mockup:

![image](https://user-images.githubusercontent.com/9599/183126312-aecfe6d4-841e-4b52-8f47-92fb599fc063.png)

**Comment by simonw** [comment]
OK, the first set of tags can be tried out here: https://datasette.io/content/plugin_repos?_facet_array=tags

<img width="936" alt="image" src="https://user-images.githubusercontent.com/9599/183127440-471e454b-3a17-4b0f-b25c-787fa3f1e774.png">

**Comment by simonw** [comment]
My first attempt at the CSS - needs a lot of refining.

<img width="765" alt="image" src="https://user-images.githubusercontent.com/9599/183129522-fbbdbb5e-74f9-4609-af0a-9a28ad47cd08.png">

```css
.tag {
  border-radius: 3px;
  white-space: nowrap;
  line-height: 2;
  border: 1px solid blue;
  padding: 3px 6px;
}
```
```html
<div style="border-bottom: 4px solid #0e0c82">
    <a class="tag" href="#">Geospatial <span>10</span></a>
    <a class="tag" href="#">Render Cell <span>9</span></a>
    <a class="tag" href="#">Output <span>8</span></a>
    <a class="tag" href="#">SQL Functions <span>8</span></a>
    <a class="tag" href="#">Authentication <span>7</span></a>
    <a class="tag" href="#">Maps <span>7</span></a>
    <a class="tag" href="#">Data Import <span>6</span></a>
    <a class="tag" href="#">API <span>5</span></a>
    <a class="tag" href="#">Visualization <span>5</span></a>
    <a class="tag" href="#">Leaflet <span>4</span></a>
    <a class="tag" href="#">Search <span>3</span></a>
    <a class="tag" href="#">Publish <span>2</span></a>
    <a class="tag" href="#">Template <span>2</span></a>
</div>
```

**Comment by simonw** [comment]
The tags need to be displayed on the plugin detail panels too.

**Comment by simonw** [comment]
I'd like to find a better name than "Render Cell" for plugins that use the `render_cell` plugin hook to customize how cells are displayed in tables.

**Comment by davidbgk** [comment]
You probably want a `tag-selected` concept too, once you click/select a tag.

Considering the style, maybe surrounding the number of elements like for instance https://getbootstrap.com/docs/5.2/components/badge/#buttons would be pertinent for readability?

**Comment by simonw** [comment]
Yes, definitely want that kind of treatment for numbers - and a toggle state for when a tag is selected.

**Comment by simonw** [comment]
One of the tags could mark known compatibility with https://lite.datasette.io/ - see:
- https://github.com/simonw/datasette-lite/issues/37

Ditto for Datasette Desktop.

**Comment by simonw** [comment]
Before adding those new tags:

<img width="181" alt="image" src="https://user-images.githubusercontent.com/9599/187513336-bed6080e-c21a-475e-a3b0-91368fdefea2.png">

After:

<img width="189" alt="image" src="https://user-images.githubusercontent.com/9599/187523151-b567edde-9e6b-48bf-b728-84726faf899d.png">
